src/index.js

```diff
@@ -38,72 +28,52 @@ class HaStore extends EventEmitter {
     this.queue = queue(
       this.config,
       this,
-      this.store,
+      this.store
     );
   }
 
-  /**
-   * Gets a list of records from source
-   * @param {string|number|array<string|number>} ids The id of the record to fetch
-   * @param {object} params (Optional)The Request parameters
-   * @returns {Promise} The eventual single record
-   */
-  async get(ids, params = {}, agg = null) {
+  get(ids, params = {}, agg = null) {
+    if (params === null) params = {};
+    const key = contextKey(this.config.delimiter, params);
+    return this.queue.getHandles(key, [ids], params, agg)
+      .then(handles => handles[0]);
+  }
+
+  getMany(ids, params = {}, agg = null) {
     if (params === null) params = {};
-    const requestIds = (Array.isArray(ids)) ? ids : [ids];
-    const key = contextKey(this.config.uniqueParams, params);
-    const handles = await this.queue.getHandles(key, requestIds, params, agg);
-    return Promise.all(handles)
-      .then(response => (!Array.isArray(ids)) ? response[0] : response);
+    const key = contextKey(this.config.delimiter, params);
+    return this.queue.getHandles(key, ids, params, agg)
+      .then((handles) => Promise.allSettled(handles)
+        .then((responses) => ids.reduce((acc, curr, index) => {
+          acc[curr] = responses[index];
+          return acc;
```

Comment:
[⚪] Context in variable names
```suggestion
        .then((outcomes) => ids.reduce((handles, id, index) => {
          handles[id] = outcomes[index];
          return handles;
```